client/homebrew/utils/versionHistory.js

```diff
@@ -43,74 +50,110 @@ function updateStoredBrew(brew, slot = 0) {
+	await IDB.getMany(historyKeys, await createHBStore())
+				.then((dataArray)=>{
+					return dataArray.forEach((data)=>{
+						history.push(data ?? DEFAULT_HISTORY_ITEM);
+					});
+				})
+				.catch(()=>{
+					historyKeys.forEach(()=>{
+						history.push(DEFAULT_HISTORY_ITEM);
+					});
+				});
+
 	return history;
 }
 
-export function updateHistory(brew) {
-	const history = loadHistory(brew);
+export async function updateHistory(brew) {
+	const history = await loadHistory(brew);
 
 	// Walk each version position
-	for (let slot = HISTORY_SLOTS; slot > 0; slot--){
+	for (let slot = HISTORY_SLOTS - 1; slot >= 0; slot--){
 		const storedVersion = history[slot];
 
 		// If slot has expired, update all lower slots and break
 		if(new Date() >= new Date(storedVersion.expireAt)){
-			for (let updateSlot = slot - 1; updateSlot>0; updateSlot--){
+
+			// Create array of arrays : [ [key1, value1], [key2, value2], ..., [keyN, valueN] ]
+			// to pass to IDB.setMany
+			const historyUpdate = [];
+
+			for (let updateSlot = slot; updateSlot > 0; updateSlot--){
 				// Move data from updateSlot to updateSlot + 1
-				!history[updateSlot]?.noData &&	updateStoredBrew(history[updateSlot], updateSlot + 1);
+				if(!history[updateSlot - 1]?.noData) {
+					historyUpdate.push(parseBrewForStorage(history[updateSlot - 1], updateSlot + 1));
+				}
 			};
 
 			// Update the most recent brew
-			updateStoredBrew(brew, 1);
+			historyUpdate.push(parseBrewForStorage(brew, 1));
+
+			IDB.setMany(historyUpdate, await createHBStore());
 
 			// Break out of data checks because we found an expired value
 			break;
 		}
 	};
 };
 
-export function getHistoryItems(brew){
-	const historyArray = [];
-
-	for (let i = 1; i <= HISTORY_SLOTS; i++){
-		historyArray.push(getVersionBySlot(brew, i));
-	}
-
-	return historyArray;
-};
-
-export function versionHistoryGarbageCollection(){
-	Object.keys(localStorage)
-		.filter((key)=>{
-			return key.startsWith(HISTORY_PREFIX);
-		})
-		.forEach((key)=>{
-			const collectAt = new Date(JSON.parse(localStorage.getItem(key)).savedAt);
-			collectAt.setMinutes(collectAt.getMinutes() + GARBAGE_COLLECT_DELAY);
-			if(new Date() > collectAt){
-				localStorage.removeItem(key);
-			}
+export async function versionHistoryGarbageCollection(){
+	if(global.config?.history?.GARBAGE_COLLECT_DELAY != GARBAGE_COLLECT_DELAY) GARBAGE_COLLECT_DELAY = global.config?.history?.GARBAGE_COLLECT_DELAY;
+
+	await IDB.entries(await createHBStore())
+		.then((entries)=>{
+			entries.forEach(async (entry)=>{
+				const key = entry[0];
+				const value = entry[1];
+
+				// if(key.startsWith(`${HISTORY_PREFIX}`)) {	// This check was to make sure we were only selecting the history keys, should be unnecessary
+				const collectAt = new Date(value.savedAt);
+				collectAt.setMinutes(collectAt.getMinutes() + GARBAGE_COLLECT_DELAY);
+				if(new Date() > collectAt){
+					IDB.del(key, await createHBStore());
+				};
+				// }
+			});
 		});
```

Comment:
The beauty of `await` is we can avoid `.then` and read things in a more traditional sequence. Bonus, is it tends to be quite a bit shorter too.

```suggestion
	const entries = await IDB.entries(await createHBStore());

	for (const [key, value] of entries) {
		const collectAt = new Date(value.savedAt);
		collectAt.setMinutes(collectAt.getMinutes() + GARBAGE_COLLECT_DELAY);

		if (new Date() > collectAt)
			await IDB.del(key, await createHBStore());
	}
```

Reply:
This is implemented in the latest commit.